src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -0,0 +1,774 @@
+                        await renewLeasesCommand.ExecuteNonQueryAsync(token);
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                // This catch block is necessary so that the finally block is executed even in the case of an exception
+                // (see https://docs.microsoft.com/dotnet/csharp/language-reference/keywords/try-finally, third
+                // paragraph). If we fail to renew the leases, multiple workers could be processing the same change
+                // data, but we have functionality in place to deal with this (see design doc).
+                this._logger.LogError($"Failed to renew leases due to error: {e.Message}");
+            }
+            finally
+            {
+                if (this._state == State.ProcessingChanges)
+                {
+                    // Do we want to update this count even in the case of a failure to renew the leases? Probably,
+                    // because the count is simply meant to indicate how much time the other thread has spent processing
+                    // changes essentially.
+                    this._leaseRenewalCount += 1;
+
+                    // If this thread has been cancelled, then the _cancellationTokenSourceExecutor could have already
+                    // been disposed so shouldn't cancel it.
+                    if (this._leaseRenewalCount == MaxLeaseRenewalCount && !token.IsCancellationRequested)
+                    {
+                        this._logger.LogWarning("Call to execute the function (TryExecuteAsync) seems to be stuck, so it is being cancelled");
+
+                        // If we keep renewing the leases, the thread responsible for processing the changes is stuck.
+                        // If it's stuck, it has to be stuck in the function execution call (I think), so we should
+                        // cancel the call.
+                        this._cancellationTokenSourceExecutor.Cancel();
+                        this._cancellationTokenSourceExecutor = new CancellationTokenSource();
+                    }
+                }
+
+                // Want to always release the lock at the end, even if renewing the leases failed.
+                this._rowsLock.Release();
+            }
+        }
+
+        /// <summary>
+        /// Resets the in-memory state of the change monitor and sets it to start polling for changes again.
+        /// </summary>
+        /// <param name="error">
+        /// The error messages the logger will report describing the reason function execution failed (used only in the case of a failure).
+        /// </param>
+        /// <param name="acquireLock">True if ClearRowsAsync should acquire the "_rowsLock" (only true in the case of a failure)</param>
+        private async Task ClearRowsAsync(string error, bool acquireLock)
+        {
+            if (acquireLock)
+            {
+                this._logger.LogError(error);
```

Comment:
Just move the logging to before the ClearRowsAsync call - the only time we pass in a string is when acquireLock is true anyways so there's no reason to have that done in here.

Reply:
Will fix this. This is probably done to save the number of locations where error is logged and `acquireLock` variable has nothing to do with logging.